lib/python/picongpu/extra/plugins/data/radiation.py

```diff
@@ -17,69 +17,65 @@
         """
-        Open references to hdf5 file to access radiation data.
+        Open references to an openPMD-api series or file to access radiation data.
 
-        This constructor opens h5py references to the radiation data
+        This constructor opens openPMD-api references to the radiation data
         and thus allows easy access to the complex amplitudes from the
         Lienard-Wiechert potential.
 
         Key-Argument:
         filename: string
-                  path and name of the hdf5 radiation data file
-
+                  path and name of the openPMD-api series or file
+        timestep: int
+                  PIC iteration to read radiation data from
         """
-        # set hdf5 file
+        # set openPMD-api file
         self.filename = filename
-        self.h5_file = h5py.File(filename, "r")
+        self.rad_series = io.Series(filename, io.Access_Type.read_only)
+
         # extract time step
-        self.timestep = self.get_timestep()
+        self.timestep = timestep
+        if (not self.timestep in self.rad_series.iterations):
+            raise Exception("The selected timestep {} ".format(self.timestep) +
+                            "is not available in the series.")
+        self.iteration = self.rad_series.iterations[self.timestep]
 
         # Amplitude
-        detectorAmplitude = self.h5_file[("/data/{}/DetectorMesh/" +
-                                          "Amplitude").format(self.timestep)]
+        detectorAmplitude = self.iteration.meshes["Amplitude"]
         # A_x
-        self.h5_Ax_Re = detectorAmplitude["x_Re"]
-        self.h5_Ax_Im = detectorAmplitude["x_Im"]
+        self.Ax_Re = detectorAmplitude["x_Re"].load_chunk()
+        self.Ax_Im = detectorAmplitude["x_Im"].load_chunk()
         # A_y
-        self.h5_Ay_Re = detectorAmplitude["y_Re"]
-        self.h5_Ay_Im = detectorAmplitude["y_Im"]
+        self.Ay_Re = detectorAmplitude["y_Re"].load_chunk()
+        self.Ay_Im = detectorAmplitude["y_Im"].load_chunk()
         # A_z
-        self.h5_Az_Re = detectorAmplitude["z_Re"]
-        self.h5_Az_Im = detectorAmplitude["z_Im"]
+        self.Az_Re = detectorAmplitude["z_Re"].load_chunk()
+        self.Az_Im = detectorAmplitude["z_Im"].load_chunk()
 
         # conversion factor for spectra from PIC units to SI units
-        self.convert_to_SI = detectorAmplitude["x_Re"].attrs['unitSI']
-
-    def get_timestep(self):
-        """Returns simulation timestep of the hdf5 data."""
-        # this is a workaround till openPMD is implemented
-        str_timestep = self.filename.split("_")[-4]
-        if str_timestep.isdigit():
-            return int(str_timestep)
-        else:
-            raise Exception("Could not extract timestep from " +
-                            "filename (\"{}\") - ".format(self.filename) +
-                            "Extracted: {}".format(str_timestep))
+        self.convert_to_SI = detectorAmplitude["x_Re"].unit_SI
```

Comment:
```suggestion
        # The value for unit_SI is consistent across all Amplitude datasets
        self.convert_to_SI = detectorAmplitude["x_Re"].unit_SI
```

I don't think that we need to check this property here, but let's maybe document that we make this assumption

Reply:
Good suggestion.I added your text